XTS/WS-T/dev/src/org/jboss/jbossts/xts/recovery/participant/ba/XTSBARecoveryManager.java

```diff
@@ -24,11 +24,15 @@ public abstract class XTSBARecoveryManager {
      */
     public static XTSBARecoveryManager getRecoveryManager()
     {
+        
+        
+
         // use a lock to get the recovery manager to avoid a race with the code that sets it
         synchronized (lock) {
             // theRecoveryManager must eventually be set to a non-null value
-            while (theRecoveryManager == null) {
+            if (theRecoveryManager == null) {
```

Comment:
`lock.wait` can return for a number of reasons so you need to retest it each time it returns.

Reply:
The existing logic looks fine, what scenario do you think it doesn't cover. But if you are unsure about it and want to change it then you'd need a reproducer for the failure in order to verify whatever change you introduce.

Reply:
But I do notice that can improve things by applying the "Double-Checked Locking" strategy. Will you update your PR to do that.

Reply:
Thanks for the review @mmusgrov ! I updated the PR implementing the "Double-Checked Locking" strategy.